File: php/src/Base/Commons/Displayable.php
<?php

namespace Shapes\Base\Commons;

trait Displayable {
    public function shouldDisplay(): bool {
        return TRUE;
    }

    public function display(): void {
        if ($this->shouldDisplay()) {
            echo "Yes, this shape can be displayed." . PHP_EOL;
        } else {
            echo "No, this shape cannot be displayed." . PHP_EOL;
        }
    }
}

File: rust/src/base/commons/displayable.rs
pub trait Displayable {
    fn should_display(&self) -> bool {
        true
    }

    fn display(&self) {
        if self.should_display() {
            println!("Yes, this shape can be displayed.");
        } else {
            println!("Shape is not displayable.");
        }
    }
}
